chemprop/uncertainty/calibrator.py

```diff
@@ -1,21 +1,65 @@
-from abc import abstractmethod
+from abc import ABC, abstractmethod
+from typing import Self
 
 from torch import Tensor
 
 from chemprop.utils.registry import ClassRegistry
 
 
-class UncertaintyCalibrator:
+class UncertaintyCalibrator(ABC):
+    """
+    A class for calibrating the predicted uncertainties.
+    """
+
     @abstractmethod
-    def fit(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> None:
+    def fit(self, preds: Tensor, uncs: Tensor, targets: Tensor, mask: Tensor) -> Self:
         """
         Fit calibration method for the calibration data.
+
+        .. note::
+            The `preds` variable is required only for regression tasks.
```

Comment:
It is difficult to make `preds` optional when it is the first positional argument. It seems simplest to break the calibrators and evaluators into three ABC's each, one for regression, one for binary classification, and one for multiclass classification. The regression and classification version would have different signatures for `fit`. In the CLI we would check the task type to know the signature and a notebook user would know what their task type is. 
@davidegraff, do you have any opinions about having three ABCs instead of just one?

Reply:
I don't really see what this current abstraction buys us if the signature expects different inputs for different implementations tbh